tests/test_plugins.py

```diff
@@ -28,6 +28,32 @@ def some_method_here(self):
     assert not checkpoint.called
 
 
+def test_registers_on_kwargs():
+
+    @slash.plugins.active  # pylint: disable=unused-variable
+    class SamplePlugin(PluginInterface):
+
+        def get_name(self):
+            return 'sample'
+
+        @plugins.registers_on('some.hook', provides=['provided_requirement'], needs=['some_requirement'], tags=['tag'])
+        @plugins.needs('other_requirement')
+        @plugins.provides('another_provided_requirement')
+        def plugin_method(self):
+            pass
+
+
+    @gossip.register('some.hook', provides=['some_requirement', 'other_requirement'])
+    def _unused():
+        pass
+
+    gossip.trigger('some.hook')
+    hook = gossip.get_hook('some.hook')
+    [r] = [r for r in hook.get_registrations() if r.func.__name__ == 'plugin_method']
```

Comment:
Can we please change `r` to `reg` or something else? (`r` is return in pdb)

Reply:
Do we want to check the `registers_on` plugin with `plugins.provides` and `plugins.needs` in class level (of the plugin) works as well?